db/db_impl.cc

```diff
@@ -4953,12 +4974,21 @@ Status DBImpl::SwitchMemtable(ColumnFamilyData* cfd, WriteContext* context) {
       }
     }
   }
+
+  // Set current_memtble_info for memtable sealed callback
+  MemTableInfo memtable_info;
+  memtable_info.cf_name = cfd->GetName();
+  memtable_info.smallest_seqno = cfd->mem()->GetFirstSequenceNumber();
+
   cfd->mem()->SetNextLogNumber(logfile_number_);
   cfd->imm()->Add(cfd->mem(), &context->memtables_to_free_);
   new_mem->Ref();
   cfd->SetMemtable(new_mem);
   context->superversions_to_free_.push_back(InstallSuperVersionAndScheduleWork(
       cfd, new_superversion, mutable_cf_options));
+
+  // Notify will drop and reacquire dbmutex
+  NotifyOnMemTableSealed(cfd, memtable_info);
```

Comment:
Thanks for your comment Igor. 

I thought about that. I would have to assume that there is no failable operation done after lock is acquired (since we would have already notified the client). This is true today. However, would this be a fair assumption to make for the future?